src/java.base/share/classes/java/lang/invoke/BootstrapMethodInvoker.java

```diff
@@ -129,25 +138,27 @@ else if (info.getClass() == int[].class) {
                 Object[] argv = (Object[]) info;
 
                 MethodType bsmType = bootstrapMethod.type();
-                if (isLambdaMetafactoryIndyBSM(bsmType) && argv.length == 3) {
+                if (bsmType == LMF_INDY_MT) {
                     result = (CallSite)bootstrapMethod
                             .invokeExact(caller, name, (MethodType)type, (MethodType)argv[0],
                                     (MethodHandle)argv[1], (MethodType)argv[2]);
-                } else if (isLambdaMetafactoryCondyBSM(bsmType) && argv.length == 3) {
-                    result = bootstrapMethod
-                            .invokeExact(caller, name, (Class<?>)type, (MethodType)argv[0],
-                                    (MethodHandle)argv[1], (MethodType)argv[2]);
-                } else if (isStringConcatFactoryBSM(bsmType) && argv.length >= 1) {
+                } else if (argv.length >= 1 && bsmType == SCF_MT) {
```

Comment:
It seems that these `argv.length` comparisons were here for cases where the wrong number of arguments are passed to a bootstrap method?

I don't see why check `argv.length > 1` is still required here... Some of the other cases are also happy to just access `argv`, (which we also do here on the next line).